pkg/granted/sso.go

```diff
@@ -155,13 +179,14 @@ var PopulateCommand = cli.Command{
 
 		g := awsconfigfile.Generator{
 			Config:              config,
-			ProfileNameTemplate: c.String("profile-template"),
-			NoCredentialProcess: c.Bool("no-credential-process"),
-			Prefix:              c.String("prefix"),
+			ProfileNameTemplate: profileNameTemplate,
+			NoCredentialProcess: noCredentialProcess,
+			Prefix:              prefix,
 			PruneStartURLs:      pruneStartURLs,
 		}
 
-		for _, s := range c.StringSlice("source") {
+		sources := coalesce.StringSlice(c.StringSlice("source"), cfg.Sso.Source)
```

Comment:
nit: could you please vendor this function into the current package so that we can avoid the additional package dependency? (the same as what you have done with `coalesceString()`)

Reply:
Another thought I had after submitting, should the command line append vs replace the list. I don't use sources, so would love your input.
`sources := append(c.StringSlice("source"), cfg.Sso.Source...)`

Reply:
Great question, some context on this is that we added `sources` to allow [Glide](https://github.com/common-fate/glide) users to call this command (`--source commonfate` would use Glide). To my knowledge it isn't widely used and we should deprecate `--source commonfate` because Glide is being sunset. This leaves `--source aws-sso` as the only remaining source flag.

Upon some further thought, what I think we should do here is remove the `Source` field that you've added to the SSO config for now:

```diff
Sso struct {
		// add any global configuration to sso profile commands defaults here.
		StartURL            string
		SsoRegion           string
		Prefix              string
-		Source              []string
		NoCredentialProcess bool
		ProfileTemplate     string
	}
```

We'll deprecate the `--source` flag and default to using AWS SSO.

In future if we add additional sources again, we can re-introduce it in the config (and this will be a non-breaking change, as the default source will remain to be AWS SSO).

Reply:
It's unlikely that we'll add additional custom sources in future here too -- we're planning on extending the [Profile Registry API](https://docs.commonfate.io/granted/usage/profile-registry) to be more flexible, rather than using this command.

Reply:
Thank you for the detail explanation.  I will keep a lookout for the extended profile registry features.  We are planning on using this heavily.